### PR TITLE
Some bugfixes and a new feature

### PR DESCRIPTION
**Pull request by 71zenith**
1) Made so that if the script fails to update for some reason, it
       will tell the user
    2) Fixed a bug which won't print version if an argument isn't
       provided
    3) Made so that the program will only try to get the url if
       neccessary(E.G. not when the user prints the help menu)
     ** EXTRA DEPENDENCY ADDED(TAC)**
    4) Made so that history will be shown reverse so the last seen anime
       will be at the top
    5) Made so that if an anime is watched with it already being on the
       history, the script will delete the line and append to the end

### CODE REVIEW

**Comment by 71zenith** [comment]
This is my first time making any pull request. So please don't mind if you see any mistake. Do tell me though

File: ani-cli [hunk 1]
```diff
@@ -64,8 +63,8 @@ update_script () {
 	if [ -z "$update" ]; then
 		printf "$c_green%s$c_reset\n" "Script is up to date :)"
 	else
-		printf '%s\n' "$update" | patch "$0" -
-		printf "$c_green%s$c_reset\n" "Script has been updated"
+		printf '%s\n' "$update" | patch "$0" - && printf "$c_green%s$c_reset\n" "Script has been updated" ||
+            printf "$c_red%s$c_reset\n" "Can't update for some reason!"
```
Comment (ura43) [inline ani-cli:67]: Sure, that seems like a good idea

File: ani-cli [hunk 2]
```diff
@@ -377,7 +376,7 @@ quality=best
 scrape=query
 download_dir="."
 
-while getopts 'viq:dp:chDUV:' OPT; do
+while getopts 'viq:dp:chDUV' OPT; do
```
Comment (ura43) [inline ani-cli:379]: I thought we needed that to catch the anime query, but turns out we don't. Nice catch.

File: ani-cli [hunk 3]
```diff
@@ -444,14 +443,16 @@ shift $((OPTIND - 1))
...
+        base_url=$(curl -s -L -o /dev/null -w "%{url_effective}\n" https://gogoanime.cm)
 		get_search_query "$*"
 		search_results=$(search_anime "$query")
 		[ -z "$search_results" ] && die "No search results found"
 		anime_selection "$search_results"
 		episode_selection
 		;;
 	history)
-		search_results=$(sed -n -E 's/\t[0-9]*//p' "$logfile")
+        base_url=$(curl -s -L -o /dev/null -w "%{url_effective}\n" https://gogoanime.cm)
```
Comment (ura43) [inline ani-cli:454]: Please move the base_url assignment before the case statement

**Review by ura43 — CHANGES_REQUESTED** [review]
While this PR has some good things, I don't see tac being useful enough to warrant its inclusion

**Comment by ura43** [comment]
Oh, also bump the version to 1.1.2

File: ani-cli (hunk 3 above)
Comment (71zenith) [reply ani-cli]: Did everything

**Review by ura43 — APPROVED** [review]

**Pull request merged**